### ISSUE TITLE
Latest released tide (tide-20160706.731) fails with "Error processing request. No Project."

### ISSUE THREAD

**Issue by notnarb** [ananthakumaran/tide#43]
Environment: Ubuntu 16.04, Emacs 24.5.1, tide-20160706.731

Steps to reproduce:
```bash
$ mkdir tide-test && cd tide-test
$ echo {} > package.json
$ echo {} > tsconfig.json # alternatively: echo '{"files":["main.ts"]}'
$ npm install --save typescript@2.0.0-dev.20160707
$ echo "const a: number = 100; console.log(a)" > main.ts
$ node_modules/.bin/tsc
$ node main.js
100
$ emacs main.ts
```
An error pops up.  Log (log level debug) included below
```json
Info 0    request: {"command":"syntacticDiagnosticsSync","seq":"1","arguments":{"file":"\/home\/bhorsley\/Workspace\/tide-test\/main.ts"}}
Err 1     Exception on executing command {"command":"syntacticDiagnosticsSync","seq":"1","arguments":{"file":"\/home\/bhorsley\/Workspace\/tide-test\/main.ts"}}:
No Project.
Error: No Project.
    at Errors (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59651:32)
    at ts.server.ts.server (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59653:11)
    at ts (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60648:7)
    at Object.<anonymous> (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60649:3)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
Info 2    response: {"seq":0,"type":"response","command":"syntacticDiagnosticsSync","request_seq":"1","success":false,"message":"Error processing request. No Project.\nError: No Project.\n    at Errors (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59651:32)\n    at ts.server.ts.server (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59653:11)\n    at ts (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60648:7)\n    at Object.<anonymous> (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60649:3)\n    at Module._compile (module.js:410:26)\n    at Object.Module._extensions..js (module.js:417:10)\n    at Module.load (module.js:344:32)\n    at Function.Module._load (module.js:301:12)\n    at Function.Module.runMain (module.js:442:10)\n    at startup (node.js:136:18)"}
Info 3    request: {"command":"semanticDiagnosticsSync","seq":"2","arguments":{"file":"\/home\/bhorsley\/Workspace\/tide-test\/main.ts"}}
Err 4     Exception on executing command {"command":"semanticDiagnosticsSync","seq":"2","arguments":{"file":"\/home\/bhorsley\/Workspace\/tide-test\/main.ts"}}:
No Project.
Error: No Project.
    at Errors (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59651:32)
    at ts.server.ts.server (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59653:11)
    at ts (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60648:7)
    at Object.<anonymous> (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60649:3)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
Info 5    response: {"seq":0,"type":"response","command":"semanticDiagnosticsSync","request_seq":"2","success":false,"message":"Error processing request. No Project.\nError: Noerror  Project.\n    at Errors (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59651:32)\n    at ts.server.ts.server (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:59653:11)\n    at ts (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60648:7)\n    at Object.<anonymous> (/home/bhorsley/dotfiles/.emacs.d/.cask/24.5.1/elpa/tide-20160706.731/tsserver/tsserver.js:60649:3)\n    at Module._compile (module.js:410:26)\n    at Object.Module._extensions..js (module.js:417:10)\n    at Module.load (module.js:344:32)\n    at Function.Module._load (module.js:301:12)\n    at Function.Module.runMain (module.js:442:10)\n    at startup (node.js:136:18)"}
```

Starting a local copy of tsserver myself is causing the same exception when issuing the same command so I believe the problem is with the command that tide is sending?

```bash
$ node_modules/.bin/tsserver 
{"command":"syntacticDiagnosticsSync","seq":"1","arguments":{"file":"\/home\/bhorsley\/Workspace\/tide-test\/main.ts"}}
Content-Length: 865

{"seq":0,"type":"response","command":"syntacticDiagnosticsSync","request_seq":"1","success":false,"message":"Error processing request. No Project.\nError: No Project.\n    at Errors (/home/bhorsley/Workspace/tide-test/node_modules/typescript/lib/tsserver.js:49798:32)\n    at ts.server.ts.server (/home/bhorsley/Workspace/tide-test/node_modules/typescript/lib/tsserver.js:49800:11)\n    at ts (/home/bhorsley/Workspace/tide-test/node_modules/typescript/lib/tsserver.js:50774:7)\n    at Object.<anonymous> (/home/bhorsley/Workspace/tide-test/node_modules/typescript/lib/tsserver.js:50775:3)\n    at Module._compile (module.js:410:26)\n    at Object.Module._extensions..js (module.js:417:10)\n    at Module.load (module.js:344:32)\n    at Function.Module._load (module.js:301:12)\n    at Module.require (module.js:354:17)\n    at require (internal/module.js:12:17)"}
```

My knowledge of tsserver commands is limited so I don't know what further debugging I can do on this, but if there's any other info I can provide, let me know!

**Comment by ananthakumaran** [comment]
I am not able to reproduce the issue
```
Info 0    request: {"command":"open","seq":"1","arguments":{"file":"\/Users\/ananthakumaran\/work\/tide-test\/main.ts","scriptKindName":"TS"}}
Info 1    Search path: /Users/ananthakumaran/work/tide-test
Info 2    Config file name: /Users/ananthakumaran/work/tide-test/tsconfig.json
Err 3     Add recursive watcher for: /Users/ananthakumaran/work/tide-test
Info 4    Opened configuration file /Users/ananthakumaran/work/tide-test/tsconfig.json
Info 5    Project (configured) 0
/Users/ananthakumaran/work/repos/tide/tsserver/lib.d.ts
/Users/ananthakumaran/work/tide-test/main.ts

-----------------------------------------------
Open file roots of inferred projects:
Open files referenced by inferred or configured projects:
Open file roots of configured projects:
/Users/ananthakumaran/work/tide-test/main.ts
Perf 6    Async elapsed time (in milliseconds): 405.5018
Info 7    request: {"command":"configure","seq":"2","arguments":{"hostInfo":"GNU Emacs 24.5.1 (x86_64-apple-darwin14.3.0, NS apple-appkit-1347.57)\n of 2015-06-18 on Ananthas-MacBook-Pro.local","file":"\/Users\/ananthakumaran\/work\/tide-test\/main.ts","formatOptions":{"tabSize":8,"indentSize":4,"insertSpaceAfterFunctionKeywordForAnonymousFunctions":true,"placeOpenBraceOnNewLineForFunctions":true}}}
Info 8    Host configuration update for file /Users/ananthakumaran/work/tide-test/main.ts
Info 9    response: {"seq":0,"type":"response","command":"configure","request_seq":"2","success":true}
Perf 10   Async elapsed time (in milliseconds): 1.2394
Info 11   request: {"command":"syntacticDiagnosticsSync","seq":"3","arguments":{"file":"\/Users\/ananthakumaran\/work\/tide-test\/main.ts"}}
Perf 12   Elapsed time (in milliseconds): 1.9531
Info 13   response: {"seq":0,"type":"response","command":"syntacticDiagnosticsSync","request_seq":"3","success":true,"body":[]}
Info 14   request: {"command":"semanticDiagnosticsSync","seq":"4","arguments":{"file":"\/Users\/ananthakumaran\/work\/tide-test\/main.ts"}}
Perf 15   Elapsed time (in milliseconds): 15.6629
```

Can you post the full log?, I want to see whether something goes wrong during the `open` or `configure` command. FYI, I am on mac

**Comment by notnarb** [comment]
Thanks for the fast response!

I believe that was my entire log, but I was experiencing this problem on my work computer and won't be able to double check until tomorrow.  It seems peculiar to me that your "Info 0" statement (seq: 1) was to open the project and my "Info 0" (seq: 1) was to call syntacticDiagnosticsSync.

I just now spun up an ubuntu/xenial64 vagrant box to test to see if I could reproduce this bug with a fresh Ubuntu 16.04 setup and the issue did not re-occur and my tss.log looks very similar to the one you posted.

When I get back to my computer I will double check that I am logging things correctly and get back to you.

**Comment by MarcusRiemer** [comment]
I am receiving the same error as @notnarb  when saving an existing `.ts` file under a new name using `C-x C-w`. No new buffer is opened, but opening it manually using `helm-projectile` works fine.

**Comment by notnarb** [comment]
Verified my settings on my work computer, tried again, and my log came up looking the same: [tss.log.txt](https://github.com/ananthakumaran/tide/files/354865/tss.log.txt)

```elisp
(defun setup-tide-mode ()                                                                                              
  (interactive)                                                                                                        
  (setq tide-tsserver-process-environment '("TSS_LOG=-level verbose -file /tmp/tss.log"))                              
  (tide-setup)                                                                                                         
  (flycheck-mode +1)                                                                                                   
  (eldoc-mode +1)                                                                                                      
  (company-mode +1))                                                                                                   
                                                                                                                       
(use-package typescript-mode                                                                                           
  :defer t                                                                                                             
  :init                                                                                                                
  (add-hook 'typescript-mode-hook #'setup-tide-mode)                                                                   
  (add-to-list 'auto-mode-alist '("\\.tsx\\'" . web-mode))                                                             
  (add-hook 'web-mode-hook (lambda ()                                                                                  
                             (when (string-equal "tsx" (file-name-extension buffer-file-name))                         
                               (setup-tide-mode))))) 
```

Tried using helm-projectile since MarcusRiemer said that worked for him (maybe the issue is related to some package's file loading hook?), but that still results in the error for me:
<img src="https://cloud.githubusercontent.com/assets/1731591/16697958/d4b76ea0-4500-11e6-91da-e204a5414d66.png" width="256"/>
<img src="https://cloud.githubusercontent.com/assets/1731591/16697961/d93c17f0-4500-11e6-8168-69f92a4bef4b.png" width="256"/>
<img src="https://cloud.githubusercontent.com/assets/1731591/16697964/de4fe366-4500-11e6-8b6d-05b1fbeee8c3.png" width="256"/>

---
While I did not get the issue on fresh install of ubuntu with a fresh init.el as mentioned in my previous comment, I did run into the issue again on another machine of mine (Debian stable, node 10.29, probably emacs 24.5.1) which shares my [init.el](https://github.com/notnarb/dotfiles/blob/master/.emacs.d/init.el).

Next thing I am going to try is clearing out my init.el and see if it is possibly a conflict with another package or setting I have configured since that's one clear thing I can see in common with both broken setups.

**Comment by notnarb** [comment]
Found what I believe is the issue.  If I comment out my entire init.el, it does not throw the error.  I slowly enabled lines until I got the error again and found that 
```elisp
(custom-set-variables '(enable-local-variables nil))
```
appears to be the culprit.  I can reproduce the issue with just that line and the relevant ones to load tide.  That line has been in my init for at least 4 years now so I can't say for sure what it's supposed to do and whether or not the blame lies with me or not for having the line.  This is the first time where I have encountered a package which has not worked with that variable set to Nil (prior versions of tide worked fine).

I'll leave it up to @ananthakumaran as to whether or not this is an issue worth fixing.  Feel free to close if not!
